### ISSUE TITLE
Copilot not working with custom vscode.FileSystemProvider

### ISSUE THREAD

**Issue by bmingles** [microsoft/vscode-copilot-release#441]
- VS Code Version: 1.82.2
- OS Version: macOS 13.2.1
- Logs:

```
2023-10-02 12:08:14.035 [error] Activating extension GitHub.copilot failed due to an error:
2023-10-02 12:08:14.035 [error] TypeError: Cannot read properties of undefined (reading 'currentScript')
    at /Users/bingles/.vscode/extensions/github.copilot-1.121.453/prompt/node_modules/web-tree-sitter/tree-sitter.js:1:140
    at /Users/bingles/.vscode/extensions/github.copilot-1.121.453/prompt/node_modules/web-tree-sitter/tree-sitter.js:1:72506
    at /Users/bingles/.vscode/extensions/github.copilot-1.121.453/node_modules/source-map-support/source-map-support.js:625:1
    at /Users/bingles/.vscode/extensions/github.copilot-1.121.453/prompt/src/parse.ts:3:20
    at /Users/bingles/.vscode/extensions/github.copilot-1.121.453/dist/extension.js:1:315
    at Object.<anonymous> (/Users/bingles/.vscode/extensions/github.copilot-1.121.453/extension/src/extension.ts:3:47)
    at Module._compile (node:internal/modules/cjs/loader:1316:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1371:10)
    at Module.load (node:internal/modules/cjs/loader:1171:32)
    at Module._load (node:internal/modules/cjs/loader:1012:12)
    at Function.f._load (node:electron/js2c/asar_bundle:2:13377)
    at Function.c._load (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:135:5630)
    at Function.m._load (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:132:29116)
    at Function.D._load (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:99:19764)
    at Module.require (node:internal/modules/cjs/loader:1195:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Function.i [as __$__nodeRequire] (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/loader.js:5:98)
    at n.rb (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:132:30295)
    at async Promise.all (index 0)
```

Steps to Reproduce:

In custom vscode extension, implement `vscode.FileSystemProvider`. Register it in the extension via:
```typescript
vscode.workspace.registerFileSystemProvider(
  "customFs",
  customFs,
  {
    isCaseSensitive: true,
  }
)
```

Initialize a workspace for the custom fs:

```typescript
vscode.workspace.updateWorkspaceFolders(0, 0, {
  uri: vscode.Uri.parse("customFs:/"),
  name: "Some Directory",
});
```

Copilot extensions show error messages:

"Activating extension 'GitHub.copilot-chat' failed: Cannot read properties of undefined (reading 'currentScript')."

and

"Activating extension 'GitHub.copilot' failed: Cannot read properties of undefined (reading 'currentScript')."

**Comment by alexdima** [comment]
@bmingles I cannot reproduce. I have tried with the fs provider extension sample at https://github.com/microsoft/vscode-extension-samples/tree/main/fsprovider-sample with the following steps:
1. clone the extension samples repo
2. open the fsprovider sample folder
3. npm install
4. launch via F5
5. In the extension development window run the command `memfs: Setup workspace`
6. (optionally save the workspace to disk)
7. Run the command `memfs: Create Files`
8. observe that the copilot and copilot chat extensions are running without errors.

<img width="1195" alt="image" src="https://github.com/microsoft/vscode-copilot-release/assets/5047891/695cf243-9c25-4a2b-a693-21cb3e4d8184">

**Comment by bmingles** [comment]
@alexdima I also tried the sample extension, and it worked as expected. I've tried comparing it to my extension and can't yet find any fundamental differences in the implementation except that mine load files / directories asynchronously. I converted the sample to use async methods and added some delays, and it still worked so not sure what's going on.

Any idea what could cause the `TypeError: Cannot read properties of undefined (reading 'currentScript')` error? Only reference I can find is something attempting to access `document` in a non-browser scope, but no idea how that applies here.
